### PR TITLE
chore: Add CODEOWNERS file

### LINKED ISSUE #727: Add CODEOWNERS file (SDK)
Model after: https://github.com/meltano/meltano/blob/main/.github/CODEOWNERS

### PR DESCRIPTION
**Pull request by edgarrmondragon**
Closes #727

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners
+
+# Ending a path in a `/` will specify the code owners for every file
+# nested in that directory, on any level
+
+# Default owners
+* @edgarrmondragon @cjohnhanson @aaronsteers
+
+# CI/CD
+/.github/workflows @edgarrmondragon @meltano/engineering
```
Comment (WillDaSilva) [inline .github/CODEOWNERS:10]:
```suggestion
/.github/workflows/ @edgarrmondragon @meltano/engineering
```

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+# Docs (General)
+/docs/ @meltano/engineering @meltano/marketing
+/README.md @afolson @tayloramurphy @meltano/engineering @meltano/marketing
+
+# Docs (Contributing)
+/docs/CONTRIBUTING.md @afolson @tayloramurphy @meltano/engineering
+
+# Release Ops (see `/.pyproject.toml` for list of bumped files)
+/cookiecutter/tap-template/{{cookiecutter.tap_id}}/pyproject.toml @meltano/engineering
+/cookiecutter/target-template/{{cookiecutter.target_id}}/pyproject.toml @meltano/engineering
```
Comment (WillDaSilva) [inline .github/CODEOWNERS:21]: GitHub is highlighting these lines as invalid. Do they need to be quoted/backslash-escaped/etc? Or does it work despite GitHub's warning?

Comment (edgarrmondragon) [reply .github/CODEOWNERS]: @WillDaSilva Ok, GitHub doesn't want the paths quoted and the braces can't be escaped so I just used a pattern.

Comment (WillDaSilva) [reply .github/CODEOWNERS]: Good idea - works well enough.

**Pull request merged**